habitat/tasks/nav/nav.py

```diff
@@ -509,6 +509,41 @@ def update_metric(
         )
 
 
+@registry.register_measure
+class SoftSPL(SPL):
+    r"""Soft SPL
+
+    Similar to SPL with a relaxed soft-success criteria. Instead of a boolean
+    success is now calculated as 1 - (ratio of distance covered to target).
+    """
+
+    def _get_uuid(self, *args: Any, **kwargs: Any):
+        return "softspl"
+
+    def update_metric(self, episode, *args: Any, **kwargs: Any):
+        current_position = self._sim.get_agent_state().position.tolist()
+        distance_to_target = self._sim.geodesic_distance(
```

Comment:
@abhiskk we should use `DISTANCE_TO_GOAL` measure here as we do in SPL.

Reply:
`distance_to_target` became more general and works for multiple goals and view points. Here is the example from SPL metric:
https://github.com/facebookresearch/habitat-api/blob/6091d0aedd41d348824252d74119d5d0d3355b8e/habitat/tasks/nav/nav.py#L478-L510